### PR TITLE
Remarks of review of code, including some additional test cases

### PR DESCRIPTION
**Pull request by fwessels**

### CODE REVIEW

File: dare.go [hunk 1]
```diff
@@ -44,9 +44,9 @@ const (
 
 var (
 	errMissingHeader      = errors.New("dare: incomplete header")
```
Comment (harshavardhana) [inline dare.go]: why are we calling this `dare:` we should name it `aead:` ?  - top level callers might not make sense of this.

Comment (aead) [inline dare.go]: We have to think about the name... AEAD is a cryptographic term for a class/type of algorithms. Can confuse users - e.g. we use AEADs in minio/aead -> recursion :wink: 

I used the simplest name I could come up with: Data At Rest Encryption -> DARE.
We are aware of the current inconstancy of the naming...

Comment (harshavardhana) [inline dare.go]: Then lets fix it properly @aead and then we can rename them appropriately. It is not correct to add errors with a different name and then have a `package-name` different.

File: reader.go [hunk 2]
```diff
@@ -136,10 +136,9 @@ func (r *decryptedReader) Read(p []byte) (n int, err error) {
 
 func (r *decryptedReader) readHeader() error {
 	n, err := io.ReadFull(r.src, header(r.pack[:]))
-	if n > 0 && err == io.ErrUnexpectedEOF {
+	if n != headerSize && err == io.ErrUnexpectedEOF {
 		return errMissingHeader
-	}
-	if err != nil {
+	} else if err != nil {
```
Comment (aead) [inline reader.go]: Why a e `else` here?

File: reader.go [hunk 3]
```diff
@@ -156,15 +155,14 @@ func (r *decryptedReader) decrypt(dst []byte) (n int, err error) {
 	ciphertext := r.pack[headerSize : headerSize+header.Len()+tagSize]
 	n, err = io.ReadFull(r.src, ciphertext)
 	if err == io.EOF || err == io.ErrUnexpectedEOF {
-		return 0, errBadPayloadLen
-	}
-	if err != nil {
+		return 0, errPayloadTooShort
+	} else if err != nil {
```
Comment (aead) [inline reader.go]: Same, no need for `else`

File: dare.go (hunk 1 above)
Comment (aead) [inline dare.go]: There is much more to fix, README, spec - we should fix this as soon as we have a stable name... I can add an issue... See #3

File: reader.go (hunk 2 above)
Comment (fwessels) [inline reader.go]: Because you are comparing err already above.

File: reader.go (hunk 3 above)
Comment (fwessels) [inline reader.go]: Same, comparing err already above.

**Pull request merged**